test/state/state.go

```diff
@@ -26,4 +26,24 @@ type State struct {
 	// This must be derived from the Rootstore.  For example it may be a namespace within the
 	// Rootstore, or even a transaction of that namespace.
 	Store corekv.Store
+
+	Txns []corekv.Txn
+}
+
+func (s *State) Clone() *State {
+	return &State{
+		T:         s.T,
+		Ctx:       s.Ctx,
+		CtxCancel: s.CtxCancel,
+		Options:   s.Options,
+		Rootstore: s.Rootstore,
+		Store:     s.Store,
+		Txns:      append([]corekv.Txn{}, s.Txns...),
+	}
+}
+
+func (s *State) WithStore(store corekv.Store) *State {
+	newState := s.Clone()
```

Comment:
question: Why did you decide to create a cloning function when `newState := *s` would give you the same outcome. Also, this seems to be dead code.

Reply:
`Clone` is not equivalent to a pointer deref as the slice is copied in `Clone`.  I think I accidentally left this in though and will remove both added funcs here.

- [ ] Remove dead code